pkg/controller/projectclaim/projectclaimadapter.go

```diff
@@ -169,7 +171,114 @@ func (c *ProjectClaimAdapter) FinalizeProjectClaim() (ObjectState, error) {
 	return ObjectUnchanged, nil
 }
 
+func (c *ProjectClaimAdapter) CreateFakeSecret() error {
+	if !gcputil.SecretExists(c.client, c.projectClaim.Spec.GCPCredentialSecret.Name, c.projectClaim.Spec.GCPCredentialSecret.Namespace) {
+		privateKeyString, err := base64.StdEncoding.DecodeString("SS1hbS1mYWtlLXBhc3M=")
+		if err != nil {
+			return err
+		}
+		if err := c.client.Create(context.TODO(), gcputil.NewGCPSecretCR(string(privateKeyString), types.NamespacedName{Namespace: c.projectClaim.Spec.GCPCredentialSecret.Namespace, Name: c.projectClaim.Spec.GCPCredentialSecret.Name})); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *ProjectClaimAdapter) DeleteFakeSecret() error {
+	secret := &corev1.Secret{}
+	err := c.client.Get(context.TODO(), types.NamespacedName{
+		Name:      c.projectClaim.Spec.GCPCredentialSecret.Name,
+		Namespace: c.projectClaim.Spec.GCPCredentialSecret.Namespace},
+		secret,
+	)
+	if err != nil {
+		return err
+	}
+
+	err = c.client.Delete(context.TODO(), secret)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *ProjectClaimAdapter) UpdateFakeProjectClaimSpecs() (bool, error) {
+	if c.projectClaim.Spec.GCPProjectID != "fakeProjectClaim" {
+		c.projectClaim.Spec.GCPProjectID = "fakeProjectClaim"
+		c.projectClaim.Spec.GCPCredentialSecret = gcpv1alpha1.NamespacedName{
+			Name:      c.projectClaim.GetName(),
+			Namespace: c.projectClaim.GetNamespace(),
+		}
+		c.projectClaim.Spec.Region = "fakeRegion"
+		c.projectClaim.Spec.AvailabilityZones = []string{
+			"fake-az-a",
+			"fake-az-b",
+			"fake-az-c",
+		}
+		err := c.client.Update(context.TODO(), c.projectClaim)
+		if err != nil {
+			return true, err
+		}
+		return false, nil
+	}
+	return true, nil
+}
+
+func (c *ProjectClaimAdapter) UpdateFakeProjectClaimState() (bool, error) {
+	if c.projectClaim.Status.State != gcpv1alpha1.ClaimStatusReady {
+		c.projectClaim.Status.Conditions = []gcpv1alpha1.Condition{}
+		c.projectClaim.Status.State = gcpv1alpha1.ClaimStatusReady
+		err := c.client.Status().Update(context.TODO(), c.projectClaim)
+		if err != nil {
+			return true, err
+		}
+		return false, nil
+	}
+	return true, nil
+}
+
+func (c *ProjectClaimAdapter) IsProjectClaimFake() (gcputil.OperationResult, error) {
```

Comment:
The name suggests this was a function that returns some kind of a boolean, but it actually initiates all the fake processing, would probably be better to call it something like `EnsureProjectClaimFakeProcessed`?